src/typesv2.ts

```diff
@@ -0,0 +1,77 @@
+import type { Id } from './idv2.js';
+
+type OmitStrict<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
+
+export type ValueType = 'TEXT' | 'NUMBER' | 'CHECKBOX' | 'URL' | 'TIME' | 'POINT';
+
+export type ValueOptions = {
+  format?: string;
+  unit?: string;
+};
+
+export type Value = {
+  value: string;
+  options?: ValueOptions;
+};
+
+export type Entity = {
+  id: string;
+  types: string[];
+  properties: Record<string, Value>;
+};
+
+export type Relation = {
+  id: string;
+  type: string;
+  fromEntity: string;
+  toEntity: string;
+  index?: string | undefined;
+};
+
+export type CreateEntityOp = {
+  type: 'CREATE_ENTITY';
+  entity: Entity;
+};
+
+export type DeleteEntityOp = {
+  type: 'DELETE_ENTITY';
+  entity: OmitStrict<Entity, 'types'>;
+};
+
+export type CreateRelationOp = {
+  type: 'CREATE_RELATION';
+  relation: Relation;
+};
+
+export type DeleteRelationOp = {
+  type: 'DELETE_RELATION';
+  relation: Pick<Relation, 'id'>;
+};
+
+export type Op = CreateEntityOp | DeleteEntityOp | CreateRelationOp | DeleteRelationOp;
```

Comment:
We also need to support the other ops from the protobuf like `UpdateEntity` and `UnsetProperties`

`UpdateEntity` is basically just an alias for `CreateEntity`. `UnsetProperties` should have the same interface but we only need to provide a `propertyId` and don't need to provide a `value`.